### PR TITLE
MM-88 add windows build to the CI

### PR DESCRIPTION
**Pull request by abidart**
To add the Windows build and have it pass the R CMD Check in the CI I first had to:

- Update setup-r in actions workflow from v1 to v2 since v1 is deprecated
- Create a `src/Makevars.win` file using  @jdstamp `src/Makevars` file (which is in the `.gitignore` and, thus, is not in the repo) and Dirk Eddelbuettel's reponse on this thread https://stackoverflow.com/questions/32974568/r-package-with-c-c-and-openmp-how-to-make-makevars-file-under-mypackage-sr . This response was usable to find a viable path for openblas, since the one in `src/Makevars` does not work for Windows.
- Update the github actions script to prevent it from adding CRLF line endings to scripts, which would otherwise cause the R CMD check to fail (for more info see: https://msmith.de/2020/03/12/r-cmd-check-github-actions.html)

### CODE REVIEW

**Comment by jdstamp** [comment]
Can I link the PR to the ticket MM-88 by commenting here?

**Comment by jdstamp** [comment]
> Can I link the PR to the ticket [MM-88](https://lcrawlab.atlassian.net/browse/MM-88) by commenting here?

Unfortunately no, the link to the ticket hast to come from the commit message I believe.

File: src/Makevars.win [hunk 1]
```diff
@@ -0,0 +1,12 @@
+PKG_CPPFLAGS = -DARMA_64BIT_WORD=1 -I.
+PKG_CXXFLAGS = -std=gnu++11  -g -O2 -I.
+CXX = /usr/local/opt/llvm/bin/clang++
+CXX11 = clang++
+PKG_LIBS = -lz   -lz -L/usr/local/Cellar/r/4.2.1_4/lib/R/lib $(LAPACK_LIBS) -L/usr/local/opt/gcc/lib/gcc/current/gcc -L/usr/local/opt/gcc/lib/gcc/current -lgfortran -lquadmath $(BLAS_LIBS)
```
Comment (jdstamp) [inline src/Makevars.win]: Did the openMP part work before?

Comment (abidart) [reply src/Makevars.win]: Are you referring to the first version of the Makevars.win file I had? That came from Dirk Eddelbuettel's reponse on this thread https://stackoverflow.com/questions/32974568/r-package-with-c-c-and-openmp-how-to-make-makevars-file-under-mypackage-sr . I think it looked something like:

```
## CXX_STD = CXX11
## PKG_CXXFLAGS = $(SHLIB_OPENMP_CXXFLAGS)
## PKG_LIBS = $(SHLIB_OPENMP_CXXFLAGS) $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS)
```

That worked but then it was missing the methods you redefined if I understood correctly. I used what that first version was using to find BLAS in order to have the Makevars.win I made from your Makevars find BLAS.

Comment (abidart) [reply src/Makevars.win]: ☝️ if that is what you meant, I updated the Makevars.win file to use that notation instead (it seems more compact) and it still passes the tests. Please double check it before merging to see if something looks particularly off or missing (you can check the diff in the very last commit to compare both).

File: src/Makevars.win [hunk 2]
```diff
@@ -0,0 +1,12 @@
+PKG_CPPFLAGS = -DARMA_64BIT_WORD=1 -I.
+PKG_CXXFLAGS = $(SHLIB_OPENMP_CXXFLAGS)
+CXX = /usr/local/opt/llvm/bin/clang++
```
Comment (jdstamp) [inline src/Makevars.win]: In reference to the discussion/last response of DIrk Eddelbuettel's suggestion in [this](https://stackoverflow.com/questions/32974568/r-package-with-c-c-and-openmp-how-to-make-makevars-file-under-mypackage-sr) post, is there a way to not assume "clang-ANYTHING"? This seems to be a requirement for portability, i.e. CRAN.

File: src/Makevars.win [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+CXX11 = clang++
```
Comment (jdstamp) [inline src/Makevars.win]: Same here.

File: configure.ac [hunk 4]
```diff
@@ -56,4 +56,4 @@ echo "
     libs:     ${PKG_LIBS} ${BLAS_LIBS} ${LAPACK_LIBS} ${FLIBS}
 
   --------------------------------------------------
-"
+"
```
Comment (jdstamp) [inline configure.ac]: This file does not need to be committed.

**Review by jdstamp — CHANGES_REQUESTED** [review]
This is amazing! I am wondering whether it is possible to get the compiler path from environment variables R tooling to ensure portability. Did you see that anywhere?

File: src/Makevars.win (hunk 3 above)
Comment (abidart) [reply src/Makevars.win]: Did my most recent change fix the issue? However the issue might persist for MacOS/Linux since I only changed the Makevars.win.

**Review by jdstamp — APPROVED** [review]
Yes! This is perfect.

**Pull request merged**